homeassistant/components/huawei_lte/config_flow.py

```diff
@@ -89,6 +90,70 @@ async def _async_show_user_form(
             errors=errors or {},
         )
 
+    async def _async_show_reauth_form(
+        self,
+        user_input: dict[str, Any],
+        errors: dict[str, str] | None = None,
+    ) -> FlowResult:
+        return self.async_show_form(
+            step_id="reauth_confirm",
+            data_schema=vol.Schema(
+                {
+                    vol.Optional(
+                        CONF_USERNAME, default=user_input.get(CONF_USERNAME) or ""
```

Comment:
```suggestion
                        CONF_USERNAME, default=user_input.get(CONF_USERNAME, "")
```
Why `or` instead of something like this?

Reply:
> I'm not sure how to trigger the reauth flow / what triggers it,

Changing the password in the router user interface does, and should be the only thing doing it.

> but I'm wondering if one can accidentally cause this by using other ways accidentally (e.g., by spamming the device with huawei-lte-api)? Preferably the reauth is not triggered by temporary login failures caused by such, right?

Valid concern, and there's no way to be 100% certain. However, repeated attempts elsewhere using incorrect credentials do not to my knowledge make this integration receive any of the three exceptions we trigger reauth with. I would not be surprised if they caused a `LoginErrorUsernamePasswordOverrunException` to pop up -- that's what we receive when the router puts an (the) account on hiatus for a while before accepting new logins even with correct credentials -- but we are not triggering reauth on that.

Reply:
Okay, that's fine by me. I don't have a test device until some time next year, so I cannot now test this. For me this PR looks good to go :+1: